File: leetcode/1390-four-divisors/sol.py
import math


class Solution:
    def sumFourDivisors(self, nums: list[int]) -> int:
        total = 0
        for num in nums:
            divisors: list[int] = []
            for j in range(1, int(math.sqrt(num)) + 1):
                if num % j == 0:
                    divisors.append(j)
                    if j != num // j:
                        divisors.append(num // j)
            if len(divisors) == 4:
                total += sum(divisors)
        return total

File: leetcode/1390-four-divisors/sol.cpp
#include <cmath>
#include <numeric>
#include <vector>

using namespace std;

class Solution {
public:
  int sumFourDivisors(vector<int> &nums) {
    int sum = 0;

    for (int num : nums) {
      vector<int> divisors;
      int limit = sqrt(num);

      for (int j = 1; j <= limit; j++) {
        if (num % j == 0) {
          divisors.push_back(j);
          if (j != num / j) {
            divisors.push_back(num / j);
          }
        }
      }

      if (divisors.size() == 4) {
        sum += accumulate(divisors.begin(), divisors.end(), 0);
      }
    }

    return sum;
  }
};
